pkg/target/http.go

```diff
@@ -219,7 +219,7 @@ func (ht *HTTPTarget) Write(messages []*models.Message) (*models.TargetWriteResu
 			continue
 		}
 		defer resp.Body.Close()
-		if resp.StatusCode == http.StatusOK {
+		if resp.StatusCode >= 200 && resp.StatusCode < 300 {
```

Comment:
Should 3xx be treated as successful as well?  I know in the Trackers a broad range of codes are seen as successful to ensure events do not get re-tried.

Reply:
See for example: https://github.com/snowplow/snowplow-golang-tracker/blob/master/tracker/emitter.go#L219

Reply:
No, 3xx should not be treated as a success.

The go http client by default follows redirects, up to 10 times per request.  So if we get a 3xx here then it means the server has issued 10 redirects in a row.  That would be highly unexpected, it probably means the server is misconfigured, and it means our event has not landed safely in the destination.

Reply:
That's fair! Feels like we might need a follow-on ticket to think through how to handle other status code groups though as currently all 3xx, 4xx and 5xx will end up being retried forever.

It might not be worth retrying certain codes forever and certain codes might not be worth retrying at all (like if we hit 3xx requests with 10+ redirects - is it worth retrying that event to have that very expensive flow followed again and again?)...